spaceward/src/features/intents/ChangePersonModal.tsx

```diff
@@ -21,8 +21,20 @@ const ChangePersonModal = ({
 		addresses.map((address) => users.includes(address)),
 	);
 
+	const isAllSelected = useMemo(() => {
+		return selected.every(Boolean);
+	}, [selected]);
+
 	const handleSelectAll = () => {
-		setSelected(Array.from({ length: addresses.length }).map(() => true));
+		if (isAllSelected)
+			setSelected(
+				Array.from({ length: addresses.length }).map(() => false),
+			);
+		else {
+			setSelected(
+				Array.from({ length: addresses.length }).map(() => true),
+			);
+		}
```

Comment:
This should be equivalent:
```suggestion
		setSelected(Array(addresses.length).fill(!isAllSelected));
```